views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```
@@ -0,0 +1,374 @@
+            properties: {
+                pageCompleteCallback:    'eba_page_complete_callback',
+                texthelpSpeechStream:    'TexthelpSpeechStream',
+                volumeValue:             'eba_volume_value'
+            },
+            functions: {
+                barDynamicStart:         '$rw_barDynamicStart',
+                barCacheInit:            '$rw_barCacheInit',
+                barInit:                 '$rw_barInit',
+                cachePage:               '$rw_cachePage',
+                disableSpeech:           '$rw_disableSpeech',
+                enableClickToSpeak:      '$rw_enableClickToSpeak',
+                enableContinuousReading: '$rw_enableContinuousReading',
+                enableSpeech:            '$rw_enableSpeech',
+                event_pause:             '$rw_event_pause',
+                event_play:              '$rw_event_play',
+                getSpeed:                '$rw_getSpeed',
+                getVoice:                '$rw_getVoice',
+                hasReachedEnd:           '$rw_hasReachedEnd',
+                isPaused:                '$rw_isPaused',
+                isSpeaking:              '$rw_isSpeaking',
+                isTextSelectedForPlay:   '$rw_isTextSelectedForPlay',
+                isUsingMathjax:          '$rw_isUsingMathjax',
+                setSpeedValue:           '$rw_setSpeedValue',
+                setUsingMath:            '$rw_setUsingMath',
+                setUsingMathjax:         '$rw_setUsingMathjax',
+                setUsingMaths:           '$rw_setUsingMaths',
+                setVolumeValue:          '$rw_setVolumeValue',
+                stopSpeech:              '$rw_stopSpeech',
+                tagSentences:            '$rw_tagSentences',
+                userParameters:          '$rw_userParameters'
+            }
+        };
+        var speed;
+        var volume;
+
+        _.assign(options || {}, {
+            contentArea: $('body'),
```

maybe we should throw an error if the contentArea is not defined instead of default to body ?

---

I think in this case 'body' is a reasonable default. If a dev were initiating this plugin outside of the 'test' context (just a hypothetical, not that it is possible), then it would be safe to assume they want the content of the entire 'body' to be used for text to speech.